google_guest_agent/events/metadata/metadata.go

```diff
@@ -67,10 +69,28 @@ func (mp *Watcher) Run(ctx context.Context, evType string) (bool, interface{}, e
 			}
 			logger.Errorf("Error watching metadata: %s", err)
 			mp.failedPrevious = true
```

Comment:
after falling back do we still want to keep `failedPrevious = true` ?

Reply:
and for all cases?

Reply:
ah yes, if fallback succeeds we should rest it, in case of failure it wouldn't matter as it would exit and restart.

Reply:
updated `tryFallback()` to reset